site/src/pages/docs/server-graphql.mdx

```diff
@@ -0,0 +1,101 @@
+
+    @Override
+    public User get(DataFetchingEnvironment environment) throws Exception {
+        final String id = environment.getArgument("id");
+        return data.get(id);
+    }
+}
+```
+
+<Tip>
+
+Please note that this is entirely based on [graphql-java](https://www.graphql-java.com/).
+
+</Tip>
+
+## `GraphqlService`
+
+Once you've finished the implementation of the [DataFetcher](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/DataFetcher.html), you need to build a <type://GraphqlService> using
+a <type://GraphqlServiceBuilder> and add it to the <type://ServerBuilder>:
+
+```java
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+ServerBuilder sb = Server.builder();
+...
+sb.service("/graphql",
+           GraphqlService.builder()
+                         .runtimeWiring(c -> {
+                            c.type("Query",
+                                   typeWiring -> typeWiring.dataFetcher("user", new UserDataFetcher()));
+                         })
+                         .build());
+...
+Server server = sb.build();
+server.start();
+```
+
+Here [RuntimeWiring](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/idl/RuntimeWiring.html) is a specification of [DataFetcher](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/DataFetcher.html)s, [TypeResolver](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/TypeResolver.html)s, and custom [scalar](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/GraphQLScalarType.html)s that are needed to wire together a functional [GraphQLSchema](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/GraphQLSchema.html).
+
+## Blocking service implementation
+
+Armeria does not run service logic in a separate thread pool by default. If your service implementation
+requires blocking, either run the individual blocking logic in a thread pool, set
+<type://GraphqlServiceBuilder#useBlockingTaskExecutor(boolean)> so the above happens automatically for
+all service methods and lifecycle callbacks.
+
+```java
+ServerBuilder sb = Server.builder();
+sb.service("/graphql",
+           GraphqlService.builder()
+                         .runtimeWiring(c -> {
+                            c.type("Query",
+                                   typeWiring -> typeWiring.dataFetcher("user", new UserDataFetcher()));
+                         })
+                         // All service methods will be run within
+                         // the blocking executor.
+                         .useBlockingTaskExecutor(true)
+                         .build());
+```
+
+<Tip>
+
+Another way is to wrap the [DataFetcher](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/DataFetcher.html) in an [AsyncDataFetcher](https://javadoc.io/doc/com.graphql-java/graphql-java/latest/graphql/schema/AsyncDataFetcher.html).
```

Comment:
How about:
You can wrap a DataFetcher using AsyncDataFetcher] to run it asynchronously.